pyang/plugins/check_update.py

```diff
@@ -518,7 +518,7 @@ def chk_default(old, new, ctx):
         oldtype = old.search_one('type')
         if (oldtype.i_typedef is not None and
             hasattr(oldtype.i_typedef, 'i_default_str') and
-            oldtype.i_typedef.i_default_str is not None and
+            oldtype.i_typedef.i_default is not None and
```

Comment:
I invested #621 a bit, and I prefer add another condition `oldtype.i_typedef.i_default_str is ""`, but I am not sure. What do you think about?

Reply:
Thanks for having a look. My understanding is the following:

i_default - represents the default object for the typedef
i_default_str - holds the string representation of the default value but this string is only valid if i_default is defined (at least that is how I interpreted the implementation)

So checking if i_default_str is empty is not sufficient because i_default could still be None.  The proposed logic will enforce that i_default is not None before checking i_default_str.